source/secure_tunneling.c

```diff
@@ -299,6 +300,58 @@ static bool s_aws_secure_tunnel_active_stream_check(
     return true;
 }
 
+/**
+ * \internal
+ * Check if DATA message can be sent to one of active connections.
+ * \endinternal
+ */
+static bool s_aws_secure_tunnel_is_data_message_valid_for_connections(
+    const struct aws_secure_tunnel *secure_tunnel,
+    const struct aws_secure_tunnel_message_view *message_view) {
+
+    if (secure_tunnel->connections->protocol_version == 0) {
+        aws_raise_error(AWS_ERROR_IOTDEVICE_SECURE_TUNNELING_DATA_NO_ACTIVE_CONNECTION);
+        return false;
+    }
+
+    if (!s_aws_secure_tunnel_protocol_version_match_check(secure_tunnel, message_view)) {
+        aws_raise_error(AWS_ERROR_IOTDEVICE_SECURE_TUNNELING_DATA_PROTOCOL_VERSION_MISMATCH);
+        return false;
+    }
+
+    /* The V1 message will be assigned a stream ID that is currently in use, so it's always valid. */
+    if (secure_tunnel->connections->protocol_version == 1) {
```

Comment:
We should also check that stream_id != 0. It may be possible that a V1 stream start was called and then a stream reset called to shut down the stream.